src/postgres/postgresConnectionStrings.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ConnectionOptions, parse } from "pg-connection-string";
+import { ParsedConnectionString } from "../ParsedConnectionString";
+import { nonNullProp } from "../utils/nonNull";
+
+export async function parsePostgresConnectionString(connectionString: string): Promise<ParsedPostgresConnectionString> {
+    connectionString = connectionString ? connectionString.trim() : '';
+    let accountConnection: boolean = false;
+    if (!connectionString.match(/^postgres:\/\/[^\/\s]+\/[^\/\s]+$/)) {
```

Comment:
If you're going to have your own parsing logic, you need to add unit tests similar to mongoConnectionString tests. This kind of complex regexp/chatAt/index stuff is too easy to mess up an edge case. If you only rely on "pg-connection-string" (the ideal case) - we don't need our own unit tests.